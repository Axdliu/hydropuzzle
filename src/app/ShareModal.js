import React, { PureComponent } from 'react';
import { Modal, Share, TouchableOpacity, View } from 'react-native';
import EStyleSheet from 'react-native-extended-stylesheet';
import { rwidth } from '@lib/rsize';
import eventEmitterInstance from '@lib/eventEmitterInstance';
import { _ } from '@app/script';

import HText from '@components/HText';

class ShareModal extends PureComponent {
  state = {
    visible: false // false
  }

  componentDidMount () {
    eventEmitterInstance.addListener('shareModal', this.show);
  }

  componentWillUnmount () {
    eventEmitterInstance.removeAllListeners('shareModal');
  }

  render () {
    return (
      <Modal
        visible={this.state.visible}
        animationType='fade'
        transparent
        onRequestClose={this.hide}>

        <View style={styles.container}>
          <View style={styles.modal}>
            <HText style={styles.title}>
              {_('common.shareModal.title')}
            </HText>
            <HText style={styles.message}>
              {_('common.shareModal.message')}
            </HText>
            <View style={styles.buttonsContainer}>
              <TouchableOpacity onPress={this.share}>
                <HText style={styles.button}>YES</HText>
              </TouchableOpacity>
              <View style={styles.buttonSeparator} />
              <TouchableOpacity onPress={this.hide}>
                <HText style={styles.button}>NO</HText>
              </TouchableOpacity>
            </View>
          </View>
        </View>
      </Modal>
    );
  }

  show = () => this.setState({visible: true});
  hide = () => this.setState({visible: false});
  share = () => Share.share({message: 'Hydropuzzle. Can you solve the mystery? https://www.sobstel.org/hydropuzzle/'}).then(this.hide);
}

export default ShareModal;

const styles = EStyleSheet.create({
  container: {
    flex: 1,
    backgroundColor: '#000d',
    alignItems: 'center',
    justifyContent: 'center'
  },
  modal: {
    borderRadius: '0.5rem',
    backgroundColor: '#fffc',
    maxWidth: rwidth(88)
  },
  title: {
    fontSize: '1.3rem',
    fontWeight: '600',
    textAlign: 'center',
    marginTop: '2rem',
    marginBottom: '1.5rem',
    paddingHorizontal: '1rem'
  },
  message: {
    textAlign: 'center',
    marginBottom: '1.5rem',
    paddingHorizontal: '1.5rem'
  },
  buttonsContainer: {
    flexDirection: 'row',
    justifyContent: 'space-around',
    borderTopWidth: 1,
    borderColor: '#999'
  },
  buttonSeparator: {
    width: 1,
    backgroundColor: '#999'
  },
  button: {
    width: rwidth(40),
    paddingVertical: '1.25rem',
    color: '#418de3',
    textAlign: 'center',
    fontSize: '1.2rem',
    fontWeight: '500'
  }
});
