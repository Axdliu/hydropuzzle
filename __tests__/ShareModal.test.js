import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';
// import { shallow } from 'enzyme';

import ShareModal from '@app/ShareModal';

describe('HButton', () => {
  const component = (
    <ShareModal />
  );

  it('renders correctly', () => {
    expect(renderer.create(component)).toMatchSnapshot();
  });
});
