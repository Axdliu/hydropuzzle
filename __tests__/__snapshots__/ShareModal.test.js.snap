// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HButton renders correctly 1`] = `
<Modal
  animationType="fade"
  hardwareAccelerated={false}
  onRequestClose={[Function]}
  transparent={true}
  visible={false}
>
  <View
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#000d",
        "flex": 1,
        "justifyContent": "center",
      }
    }
  >
    <View
      style={
        Object {
          "backgroundColor": "#fffc",
          "borderRadius": 7.75,
          "maxWidth": 660,
        }
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "backgroundColor": "transparent",
              "color": "black",
              "fontSize": 17.05,
              "fontWeight": "400",
            },
            Object {
              "fontFamily": "System",
              "lineHeight": 23.25,
            },
            Object {
              "fontSize": 20.150000000000002,
              "fontWeight": "600",
              "marginBottom": 23.25,
              "marginTop": 31,
              "paddingHorizontal": 15.5,
              "textAlign": "center",
            },
          ]
        }
        type="default"
      >
        Cheap marketing gimmick
      </Text>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "backgroundColor": "transparent",
              "color": "black",
              "fontSize": 17.05,
              "fontWeight": "400",
            },
            Object {
              "fontFamily": "System",
              "lineHeight": 23.25,
            },
            Object {
              "marginBottom": 23.25,
              "paddingHorizontal": 23.25,
              "textAlign": "center",
            },
          ]
        }
        type="default"
      >
        This is not a part of the game, but maybe you want to tell the world about Hydropuzzle?
      </Text>
      <View
        style={
          Object {
            "borderColor": "#999",
            "borderTopWidth": 1,
            "flexDirection": "row",
            "justifyContent": "space-around",
          }
        }
      >
        <View
          accessibilityComponentType={undefined}
          accessibilityLabel={undefined}
          accessibilityTraits={undefined}
          accessible={true}
          collapsable={undefined}
          hasTVPreferredFocus={undefined}
          hitSlop={undefined}
          isTVSelectable={true}
          nativeID={undefined}
          onLayout={undefined}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "opacity": 1,
            }
          }
          testID={undefined}
          tvParallaxProperties={undefined}
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Array [
                Object {
                  "backgroundColor": "transparent",
                  "color": "black",
                  "fontSize": 17.05,
                  "fontWeight": "400",
                },
                Object {
                  "fontFamily": "System",
                  "lineHeight": 23.25,
                },
                Object {
                  "color": "#418de3",
                  "fontSize": 18.599999999999998,
                  "fontWeight": "500",
                  "paddingVertical": 19.375,
                  "textAlign": "center",
                  "width": 300,
                },
              ]
            }
            type="default"
          >
            YES
          </Text>
        </View>
        <View
          style={
            Object {
              "backgroundColor": "#999",
              "width": 1,
            }
          }
        />
        <View
          accessibilityComponentType={undefined}
          accessibilityLabel={undefined}
          accessibilityTraits={undefined}
          accessible={true}
          collapsable={undefined}
          hasTVPreferredFocus={undefined}
          hitSlop={undefined}
          isTVSelectable={true}
          nativeID={undefined}
          onLayout={undefined}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "opacity": 1,
            }
          }
          testID={undefined}
          tvParallaxProperties={undefined}
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Array [
                Object {
                  "backgroundColor": "transparent",
                  "color": "black",
                  "fontSize": 17.05,
                  "fontWeight": "400",
                },
                Object {
                  "fontFamily": "System",
                  "lineHeight": 23.25,
                },
                Object {
                  "color": "#418de3",
                  "fontSize": 18.599999999999998,
                  "fontWeight": "500",
                  "paddingVertical": 19.375,
                  "textAlign": "center",
                  "width": 300,
                },
              ]
            }
            type="default"
          >
            NO
          </Text>
        </View>
      </View>
    </View>
  </View>
</Modal>
`;
